exposed-dao/src/main/kotlin/org/jetbrains/exposed/dao/EntityCache.kt

```diff
@@ -23,6 +23,9 @@ class EntityCache(private val transaction: Transaction) {
     internal val inserts = LinkedHashMap<IdTable<*>, MutableSet<Entity<*>>>()
     private val updates = LinkedHashMap<IdTable<*>, MutableSet<Entity<*>>>()
     internal val referrers = HashMap<Column<*>, MutableMap<EntityID<*>, SizedIterable<*>>>()
+    internal val innerTableLinks by lazy {
+        HashMap<Column<*>, MutableMap<EntityID<*>, MutableSet<InnerTableLinkEntity<*>>>>()
+    }
```

Comment:
Most ideally, there should be a cache that stores the link entities without any relation to their referenced counterparts, solely based on some special id, which could be retrieved from the `ResultRow` in `wrapLinkRow()`. This is how the `data` cache is for example set up for all regular entities.
This would mean either a brand new id for the entity (defeats the purpose as the point is to not introduce a new/fake id column in the intermediate table, since uniqueness is based on the 2 referencing columns) or some way to check `ResultRow` values against entity values. For the latter, I did consider forcing another override where the user defines some sort of equality match between `ResultRow` and `InnerTableLinkEntity`, but it got a bit messy.

What the above cache does is store all `InnerTableLinkEntity`s for a target column and source (column) id, so uniqueness essentially relies on 3 values: target column (e.g. `task` in `ProjectTasks`), source id (e.g. `project` value in `ProjectTasks`), and target id (e.g. `TaskWithData.id` stored in the entity itself).